parity/account_utils.rs

```diff
@@ -152,8 +152,9 @@ mod accounts {
 				engine_signer,
 				password.clone(),
 			);
-			if signer.sign(Default::default()).is_ok() {
-				author = Some(::ethcore::miner::Author::Sealer(Box::new(signer)));
+			// NOTE: rust-secp256k1, a Message cannot be all zeroes anymore
+			if signer.sign([1_u8; 32].into()).is_ok() {
```

Comment:
Good catch. `Default::default()` is a rust idiom I really don't like, it's too clever for its own good. Can we also do something better than just fail silently, i.e. `warn!` if the dummy signing fails?

Reply:
> Default::default() is a rust idiom I really don't like, it's too clever for its own good

Agree, it is quite elegant not to have to import specific type and so on ..., but bad for readability.

We are actually printing, `No valid password for the consensus signer {}. {}", engine_signer, VERIFY_PASSWORD_HINT` when this fails in `parity::execute_impl`. Is that not sufficient or do you want to log each failed `dummy sign` attempt? It may be noisy if several passwords are used

The reported error is bad, https://github.com/OpenEthereum/open-ethereum/blob/master/rpc/src/v1/helpers/engine_signer.rs#L41.

Reply:
> We are actually printing, No valid password for the consensus signer

Yeah, it's just that it's not their password that is the problem, but the payload.

I guess it's kind of moot now that you've fixed it so let's just leave it.

I'd remove the comment though, or do you think it's useful?

Reply:
…or maybe add a `const DUMMY_SECP_MESSAGE` and slap a big comment there?

Reply:
I don't think the comment is necessary, it was more for  you reviewers.

A constant is better and self-explained, self-documenting code for the win :muscle: